action.yml

```
@@ -43,18 +49,41 @@ runs:
         [ "${{ inputs.cache-registries }}" = "true" ] && R_PATH="~/.julia/registries"
         echo "registries-path=$R_PATH" >> $GITHUB_OUTPUT
         [ "${{ inputs.cache-compiled }}" = "true" ] && PCC_PATH="~/.julia/compiled"
-        echo "precompilation-cache-path=$PCC_PATH" >> $GITHUB_OUTPUT
+        echo "compiled-path=$PCC_PATH" >> $GITHUB_OUTPUT
         [ "${{ inputs.cache-scratchspaces }}" = "true" ] && S_PATH="~/.julia/scratchspaces"
         echo "scratchspaces-path=$S_PATH" >> $GITHUB_OUTPUT
+        [ "${{ inputs.cache-logs }}" = "true" ] && L_PATH="~/.julia/logs"
+        echo "logs-path=$L_PATH" >> $GITHUB_OUTPUT
       shell: bash
 
-    - uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84
+    - uses: actions/cache@4d4ae6ae148a43d0fd1eda1800170683e9882738
       id: cache
       with:
-        path: "${{ format('{0}\n{1}\n{2}\n{3}\n{4}', steps.paths.outputs.artifacts-path, steps.paths.outputs.packages-path, steps.paths.outputs.registries-path, steps.paths.outputs.precompilation-cache-path, steps.paths.outputs.scratchspaces-path) }}"
-        key: ${{ runner.os }}-${{ inputs.cache-name }}-${{ hashFiles('**/Project.toml') }}
+        path: |
+          ${{ steps.paths.outputs.artifacts-path }}
+          ${{ steps.paths.outputs.packages-path }}
+          ${{ steps.paths.outputs.registries-path }}
+          ${{ steps.paths.outputs.scratchspaces-path }}
+          ${{ steps.paths.outputs.compiled-path }}
+          ${{ steps.paths.outputs.logs-path }}
+        # the key should be unique to the run, otherwise actions/cache will refuse to save it
+        key: ${{ inputs.cache-name }}-${{ matrix.julia-version }}-${{ runner.os }}-${{ matrix.arch }}-${{ github.run_id }}-${{ github.run_attempt }}
+        # Note: if changed, make sure the restore key is updated in the delete step too
         restore-keys: |
-          ${{ runner.os }}-${{ inputs.cache-name }}-
+          ${{ inputs.cache-name }}-${{ matrix.julia-version }}-${{ runner.os }}-${{ matrix.arch }}
```

Actually `version` might be more common than `julia-version`.

Or we just be sensitive to both like this (given non-existant matrix var is interpolated as an empty string)
```suggestion
          ${{ inputs.cache-name }}-${{ matrix.version }}${{ matrix.julia-version }}-${{ runner.os }}-${{ matrix.arch }}
```